matching-notification-api/src/main/java/org/phenotips/matchingnotification/finder/internal/AbstractMatchFinder.java

```diff
@@ -100,9 +107,7 @@
 
     private DateTimeFormatter dateFormatter = ISODateTimeFormat.dateTime().withZone(DateTimeZone.UTC);
 
-    protected abstract Set<String> getSupportedServerIdList();
-
-    protected abstract MatchRunStatus specificFindMatches(Patient patient, String serverId,
+    protected abstract Response specificFindMatches(Patient patient, String serverId,
```

Comment:
Having matching code to return an object related to web (`Response`) seems wrong. I don't have a nice a quick solution how to do this otherwise, but it adds an unnecessary dependency on `javax.ws.rs.core` to a piece of code which should not care about that. A not-perfect other solution is to have our own "status" object, with similar functionality to a `Response`, and do a conversion from that object to a Response in REST code. @marta, what do you think, which design patter is better (one of these two, or something else)?

Reply:
The design problem is very deep, I suggest the refactoring be done separately from this PR.